src/ipfsd-in-proc.js

```diff
@@ -68,6 +72,8 @@ class Node {
       EXPERIMENTAL: this.opts.EXPERIMENTAL,
       libp2p: this.opts.libp2p
     })
+
+    this.exec.once('ready', () => this.emit('ready'))
```

Comment:
Why use a different pattern than the one used for ipfs-daemon (new Node + node.start)?

Reply:
I believe not doing that can end up in race conditions - take a look at [maybeOpenRepo](https://github.com/ipfs/js-ipfs/blob/d945fceeb09c1c4289a61edb40209271e8318d25/src/core/boot.js#L67), its called by the IPFS constructore through `boot(self)`, but it is async, and the only way of knowing when it finished is by hooking into the `ready` event.

Reply:
What I'm weird'ed out is that in `ipfsd-in-proc`, the IPFS instance starts (and therefore does network activity, lock repo, etc) on the constructor, while in `ipfsd-daemon`, the daemon only starts when .start is called. This asymmetry will lead to confusion and debugging issues in the future.

Reply:
ah, I see - we're not actually starting the node since IPFS is constructed with both, `init` and `start` as false, take a look here - https://github.com/ipfs/js-ipfsd-ctl/blob/a61fa50d6834e61465398cd1d54e60ec913358eb/src/ipfsd-in-proc.js#L67...L74. This was done specifically to keep that symmetry with the other types of daemons, the `ready` event is there to make sure we don't trip over anything.

Reply:
I see. Makes sense then. Thanks for clarifying :)